File: frontend/src/dto/CVStyleDto.ts
import { CVStyleOptionType } from '@/dto/CVStyleOptionType.ts'

export interface CVStyleDto {
  key: string
  name: string
  description: string
  options: Array<CVStyleOptionDto>
}

export interface CVStyleOptionDto {
  key: string
  name: string
  type: CVStyleOptionType
  default: string
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/cv/generator/CVStyleDto.kt
package ch.streckeisen.mycv.backend.cv.generator

data class CVStyleDto(
    val key: String,
    val name: String,
    val description: String,
    val options: List<CVStyleOptionDto> = emptyList()
)

data class CVStyleOptionDto(
    val key: String,
    val name: String,
    val type: CVStyleOptionType,
    val default: String
)
